pkg/state/impl/etcd/key.go

```diff
@@ -0,0 +1,32 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package etcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+)
+
+const (
+	etcdKeyPrefix = "cosi"
+)
+
+// etcdKeyFromPointer builds a key for the resource with the given pointer.
+func etcdKeyFromPointer(pointer resource.Pointer) string {
+	prefix := etcdKeyPrefixFromKind(pointer)
+	idEscaped := url.QueryEscape(pointer.ID())
```

Comment:
do we need an ID here, or can we get away with sha256(id)?

Reply:
Nope, we can get away with `/sha256(ns)/sha256(type)/sha256(id)` or even `/sha256(ns + "_" + type)/sha256(id)` or anything like that.

Reply:
I think sha might be nice from encryption point of view... probably less sense for `type`/`ns`, but `id` might be nice. 

probably we can also add some `salt`.

Reply:
done